app/src/renderer/components/common/ActionModal.vue

```diff
@@ -126,50 +147,73 @@ export default {
     }
   },
   data: () => ({
+    step: `txDetails`,
     signMethod: null,
     password: null,
-    selectedSignMethod: `local`,
-    signMethods: [
-      {
-        key: `Account Password (Unsafe)`,
-        value: `local`
-      }
-      // {
-      //   key: `Ledger`,
-      //   value: `ledger`
-      // },
-      // {
-      //   key: `Cosmos Signer App`,
-      //   value: `signer-app`
-      // }
-    ],
     sending: false,
     submissionError: null,
     show: false
   }),
   computed: {
-    ...mapGetters([`connected`])
+    ...mapGetters([`connected`, `ledger`]),
+    selectedSignMethod() {
+      if (this.ledger.isConnected) {
+        return `ledger`
+      }
+      return `local`
+    },
+    signMethods() {
+      if (this.ledger.isConnected) {
+        return [
+          {
+            key: `Ledger`,
+            value: `ledger`
+          }
+        ]
+      }
+      return [
+        {
+          key: `(Unsafe) Local Account`,
+          value: `local`
+        }
+      ]
+    }
   },
   methods: {
     open() {
       this.show = true
     },
     close() {
       this.password = null
+      this.step = `txDetails`
       this.show = false
     },
-    async validateForm() {
-      this.sending = true
+    async validateChangeStep() {
       this.$v.$touch()
 
       // An ActionModal is only the prototype of a parent modal
       // here we trigger the validation of the form that this parent modal
       const childFormValid = this.validate()
 
       if (!this.$v.$invalid && childFormValid) {
-        await this.submit()
+        if (
+          this.selectedSignMethod === `local` ||
```

Comment:
hard to read, a lot of ifs and else-ifs

Reply:
might be nice to remove some of the logic from here and add the various states as computed properties.

Reply:
updated